homeassistant/components/v2c/diagnostics.py

```diff
@@ -0,0 +1,38 @@
+"""Diagnostics support for V2C."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.util.json import json_loads
+
+from .const import DOMAIN
+from .coordinator import V2CUpdateCoordinator
+
+TO_REDACT = {
+    CONF_HOST,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: V2CUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
```

Comment:
In a separate PR, migrate the integration to store the coordinator in entry.runtime_data instead